draft-ietf-rtcweb-jsep.xml

```diff
@@ -3013,10 +3015,10 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             and address of the default candidate for the m= section, as
             described in
             <xref target="I-D.ietf-mmusic-ice-sip-sdp"></xref>,
-            Section 3.2.1.2. Note, however,
-            that the m= line protocol need not match the default
-            candidate, because this protocol value must instead match
-            what was supplied in the offer, as described above.</t>
+            Section 3.2.1.2. Note that in certain cases, the m= line protocol
+            may not match that of the default candidate, because the m= line
+            protocol value MUST match what was supplied in the offer, as
+            described above.</t>
```

Comment:
Should we say here what address and port should be used in this case in c= and m= line? I would say that dummy values should be used for address and port.

Reply:
As noted, it has to be the IP/port of the default candidate.

Reply:
dummy values are only used if no candidates have yet been gathered.

Reply:
What is the reason for the "candidate in active use" deviation? If ICE checking has completed for multiple pairs, the endpoints can switch between those pairs. Which pair is the "active"?

Also, we should use proper ICE terminology. Instead of saying "If ICE checking has completed for multiple pairs" we should say "If ICE has produced one ore more valid pairs", or something like that...

Reply:
@cdh4u, not sure which text you are referring to here.

Reply:
The text on row 2368.

Reply:
ah, ok. The goal here is to avoid any ambiguity, I believe. It would be the most-recently-used pair for sending media.

Reply:
> ah, ok. The goal here is to avoid any ambiguity, I believe. It would be the most-recently-used pair for sending media.

I think that should be clarified. Let's not introduce new ICE terminology - is it messy enough already :) 

But, before nomination takes place, endpoints can send media on ANY valid pair, and the pair used for sending media can change. I think the text should say *WHY* JSEP deviates from draft-ice-sdp in this regard.

Reply:
One of the reasons why ice-sip-sdp specified default and not the "most-recently-used pair", was that most-recently used pair can change between offer and answer. Furthermore, candidates used by different end points during ICE nomination can belong to different candidate pairs and even used different transport (for instance if one of UDP connectivity check packets got lost due to packet loss). This results in candidates on the m= line and even protocols which do not match. Stating that default candidate MUST be one of the UDP candidates, makes c=/m= line change less often and be more consistent. Also, changing c=/m= line this often serves little purpose when both end points support ICE (as they should for JSEP).

Regarding the protocol and candidate mismatch in the answer, this can result in ICE mismatch and negotiation failure. See  https://tools.ietf.org/html/draft-ietf-mmusic-ice-sip-sdp-24#section-3.3.3 for more details. Once again, putting dummy value is safer there (but can still cause issues with older implementations).

Reply:
> One of the reasons why ice-sip-sdp specified default and not the "most-recently-used pair", was that most-recently used pair can change between offer and answer. Furthermore, candidates used by different end points during ICE nomination can belong to different candidate pairs and even used different transport (for instance if one of UDP connectivity check packets got lost due to packet loss). This results in candidates on the m= line and even protocols which do not match. Stating that default candidate MUST be one of the UDP candidates, makes c=/m= line change less often and be more consistent. Also, changing c=/m= line this often serves little purpose when both end points support ICE (as they should for JSEP).
> 
> Regarding the protocol and candidate mismatch in the answer, this can result in ICE mismatch and negotiation failure. See https://tools.ietf.org/html/draft-ietf-mmusic-ice-sip-sdp-24#section-3.3.3 for more details. Once again, putting dummy value is safer there (but can still cause issues with older implementations).

@rshpount, I agree with you, and that's why I would like to know why JSEP does things differently.